src/sales/schemas/sale.schemas.ts

```diff
@@ -0,0 +1,24 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+
+export type SaleDocument = Sale & Document;
+
+@Schema()
+export class Sale {
+    @Prop({ required: true })
+    subtotal: number;
+
+    @Prop({ required: true })
+    total: number;
+
+    @Prop({ required: true })
+    cartItem: [{
+        productId:string;
+        quantity:number;
+    }];
+
+    @Prop({ default: new Date() })
+    createdAt: Date;
+}
+
+export const SaleSchema = SchemaFactory.createForClass(Sale);
```

Comment:
Crear otro schema para el CartItem, así vas a poder manejar de manera correcta la referencia a los productos, igual debes tipar correctamente cada prop.

CartItem y CartItemSchema muevelo a otro archivo `cart-item.schema.ts` en la misma rutas 

```suggestion
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { Document, Type } from 'mongoose';
import { Product } from '../../product/schemas/product.schema';

@Schema()
export class CartItem {
    @Prop({type: Types.ObjectId, ref: Product.name, required: true})
    product: Product | Types.ObjectId
    
    @Props({type: Number, required: true})
    quantity: number
}

export const CartItemSchema = SchemaFactory.createForClass(CartItem );

@Schema()
export class Sale extends Document {
    @Prop({ type: Number, required: true })
    subtotal: number;

    @Prop({ type: Number, required: true })
    total: number;

    @Prop({ type: [CartItemSchema],  required: true })
    cart: Types.Array<CartItem> 

    @Prop({ default: new Date() })
    createdAt: Date;
}

export const SaleSchema = SchemaFactory.createForClass(Sale);
```